src/components/Articles.jsx

```diff
@@ -5,38 +5,52 @@ import ArticleStyle from "./ArticleStyle";
 import Loading from "./Loading";
 import Topics from "./Topics";
 import { useSearchParams } from "react-router-dom";
+import SortArticles from "./SortArticles";
 
 export default function Articles() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
   const [topic, setTopic] = useState("all");
   const [searchParams, setSearchParams] = useSearchParams();
-
+  const [orderQuery, setOrderQuery] = useState("DESC");
+  const [sortBy, setSortBy] = useState("created_at");
+  const [orderQueryBtn, setOrderQueryBtn] = useState("Descending ↓");
   const queryTopic = searchParams.get("topic");
 
   useEffect(() => {
     if (queryTopic) {
       setTopic(queryTopic);
     }
+  }, []);
+
+  useEffect(() => {
     if (topic !== "all") {
```

Comment:
is there a way to set a default for the topic so that if no topic is provided it just fetches all the articles? this would reduce the need to conditionally call the function :)